File: client-scalaz/src/main/scala/org/openapitools/client/api/RoutingListResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import RoutingListResponse._

case class RoutingListResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
items: Option[List[RoutingResponse]],
count: Option[Integer],
debug: Option[List[String]],
returning: Option[String])

object RoutingListResponse {
  import DateTimeCodecs._

  implicit val RoutingListResponseCodecJson: CodecJson[RoutingListResponse] = CodecJson.derive[RoutingListResponse]
  implicit val RoutingListResponseDecoder: EntityDecoder[RoutingListResponse] = jsonOf[RoutingListResponse]
  implicit val RoutingListResponseEncoder: EntityEncoder[RoutingListResponse] = jsonEncoderOf[RoutingListResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/routing_list_response.clj
(ns sirqul-io-t-platform.specs.routing-list-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.routing-response :refer :all]
            )
  (:import (java.io File)))


(def routing-list-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :items) (s/coll-of routing-response-spec)
   (ds/opt :count) int?
   (ds/opt :debug) (s/coll-of string?)
   (ds/opt :returning) string?
   })

(def routing-list-response-spec
  (ds/spec
    {:name ::routing-list-response
     :spec routing-list-response-data}))
